File: BMB2006/H5O2/Dugum.cs
/*
 *         Ferit Yiğit BALABAN, <fybalaban@fybx.dev>
 *         032190002
*/

namespace H5O2;

public class Dugum {
	public int deger;
	public Dugum sonrakiDugum;
	public Dugum oncekiDugum;

	public Dugum(int d) {
		deger = d;
		sonrakiDugum = null;
		oncekiDugum = null;
	}
}

File: BMB2006/H7O3/dugum.c
/*
 *          Ferit Yiğit BALABAN <fybalaban@fybx.dev>
 *          032190002
 *
 *          dugum.c
 */

#include <stdlib.h>
#include "dugum.h"

Dugum* yeni_dugum(int deger) {
    Dugum *d    = (Dugum *) calloc(1, sizeof(Dugum));
    d->deger    = deger;
    d->sonraki  = NULL;
    return d;
}

int dugum_guncelle(Dugum* dugum, int deger) {
    if (!dugum)
        return -1;

    dugum->deger = deger;
    return deger;
}